src/tree_sortable.rs

```diff
@@ -0,0 +1,134 @@
+// This file was generated by gir (17af302) from gir-files (11e0e6d)
+// DO NOT EDIT
+
+use SortType;
+use ffi;
+use glib::object::{Downcast, IsA};
+use glib::translate::*;
+use auto::traits::TreeSortableExt as Auto;
+use std::mem::transmute;
+use std::cmp::Ordering;
+
+use signal::CallbackGuard;
+
+use glib_ffi::gpointer;
+use {TreeSortable, TreeIter};
+use ffi::{GtkTreeIter, GtkTreeSortable};
+
+pub enum SortColumn {
+    Default,
+    Unsorted,
+    Index(u32),
+}
+
+impl ToGlib for SortColumn {
+    type GlibType = i32;
+
+    #[inline]
+    fn to_glib(&self) -> i32 {
+        match *self {
+            SortColumn::Default => ffi::GTK_TREE_SORTABLE_DEFAULT_SORT_COLUMN_ID,
+            SortColumn::Unsorted => ffi::GTK_TREE_SORTABLE_UNSORTED_SORT_COLUMN_ID,
+            SortColumn::Index(x) => {
+                assert!(x <= i32::max_value() as u32, "column index is too big");
+                x as i32
+            }
+        }
+    }
+}
+
+impl FromGlib<i32> for SortColumn {
+    #[inline]
+    fn from_glib(val: i32) -> SortColumn {
+        match val {
+            ffi::GTK_TREE_SORTABLE_DEFAULT_SORT_COLUMN_ID => SortColumn::Default,
+            ffi::GTK_TREE_SORTABLE_UNSORTED_SORT_COLUMN_ID => SortColumn::Unsorted,
+            x => {
+                assert!(x >= 0, "invalid column index");
+                SortColumn::Index(x as u32)
+            }
+        }
+    }
+}
+
+pub trait TreeSortableExt {
+    // -- manual --
+    fn set_default_sort_func<F>(&self, sort_func: F)
+    where F: Fn(&Self, &TreeIter, &TreeIter) -> Ordering + 'static;
+    fn set_sort_func<F>(&self, sort_column_id: SortColumn, sort_func: F)
+    where F: Fn(&Self, &TreeIter, &TreeIter) -> Ordering + 'static;
+    fn get_sort_column_id(&self) -> Option<(SortColumn, SortType)>;
+    fn set_sort_column_id(&self, sort_column_id: SortColumn, order: SortType);
+    // -- auto --
+    fn has_default_sort_func(&self) -> bool;
+    fn sort_column_changed(&self);
+}
+
+unsafe extern "C" fn trampoline<T>(this: *mut GtkTreeSortable, iter: *mut GtkTreeIter,
+                                   iter2: *mut GtkTreeIter,
+                                   f: &mut Box<Fn(&T, &TreeIter, &TreeIter) -> Ordering>)
```

Comment:
If we forgo closure mutability the borrow of `f` has to be immutable.

But as per an earlier comment, I'd rather the type here were `gpointer`, cast to `&Box` in the body.